crd-generator/src/main/java/io/fabric8/crd/generator/apt/CustomResourceAnnotationProcessor.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.crd.generator.apt;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator.Feature;
+import io.fabric8.crd.generator.v1.CustomResourceHandler;
+import io.fabric8.kubernetes.api.model.KubernetesList;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.utils.ApiVersionUtil;
+import io.sundr.codegen.CodegenContext;
+import io.sundr.codegen.functions.ElementTo;
+import io.sundr.codegen.model.TypeDef;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.Name;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeMirror;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+
+@SupportedAnnotationTypes({"io.fabric8.kubernetes.model.annotation.Version"})
+public class CustomResourceAnnotationProcessor extends AbstractProcessor {
+
+  private static final ObjectMapper YAML_MAPPER = new ObjectMapper(
+    new YAMLFactory()
+      .enable(Feature.MINIMIZE_QUOTES)
+      .enable(Feature.ALWAYS_QUOTE_NUMBERS_AS_STRINGS)
+      .disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER)
+  );
+  static {
+    YAML_MAPPER.configure(SerializationFeature.INDENT_OUTPUT, true);
+    YAML_MAPPER.configure(SerializationFeature.WRITE_NULL_MAP_VALUES, false);
+    YAML_MAPPER.configure(SerializationFeature.WRITE_EMPTY_JSON_ARRAYS, false);
+  }
```

Comment:
Would it make sense to reuse the Serialization#yamlMapper instance?
https://github.com/fabric8io/kubernetes-client/blob/db1923b62b44b774077cf5359b0492ab6eda75bb/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/Serialization.java#L56

Reply:
Not sure because we use a specific configuration and I wouldn't want it to impact the client.

Reply:
I don't think that this configuration might "hurt" the default client configuration. However, you're right, this is the safer approach.